### PR TITLE
patchelf.rb reading for everyone :)

### PR DESCRIPTION
**Pull request by rmNULL**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

enable `patchelf.rb` reading for everyone. 

Those who don't wish to use `patchelf.rb` can do so by 
exporting `HOMEBREW_NO_PATCHELF_RB=1` to their env.

### CODE REVIEW

**Comment by rmNULL** [comment]
cc @sjackman @woodruffw @MikeMcQuaid

File: Library/Homebrew/os/linux/global.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 # enables experimental readelf.rb, patchelf support.
```
Comment (MikeMcQuaid) [inline Library/Homebrew/os/linux/global.rb:3]: Not experimental any more. Let's just remove all usage of `HOMEBREW_PATCHELF_RB` and `HOMEBREW_NO_PATCHELF_RB` and delete the legacy code. We can revert this PR if we need to turn it off after that.

Comment (woodruffw) [reply Library/Homebrew/os/linux/global.rb:3]: Agreed with this; could you make these changes @rmNULL?

**Comment by woodruffw** [comment]
CI failure _looks_ unrelated:

> 1st Try error in ./test/cmd/services_spec.rb:4:
expected block to output "Warning: No services available to control with `brew services`\n" to stderr, but output "Warning: Calling Homebrew.args is deprecated! Use `args = <command>_args.parse` and pass `args` along the call chain instead.\nPlease report this issue to the homebrew/services tap (not Homebrew/brew or Homebrew/core), or even better, submit a PR to fix it:\n  /private/tmp/homebrew-tests-20200808-15182-19bc5zq/prefix/Library/Taps/homebrew/homebrew-services/lib/services_cli.rb:65\n\nWarning: No services available to control with `brew services`\n"

**Comment by MikeMcQuaid** [comment]
Thanks @rmNULL!

**Pull request merged**